block-storage/src/main/scala/coop/rchain/blockstorage/BlockDagFileStorage.scala

```diff
@@ -193,6 +193,15 @@ final class BlockDagFileStorage[F[_]: Concurrent: Sync: Log: BlockStore: RaiseIO
       } yield result
     def lookup(blockHash: BlockHash): F[Option[BlockMetadata]] =
       dataLookup.get(blockHash).pure[F]
+    def unsafeLookup(blockHash: BlockHash): F[BlockMetadata] =
+      dataLookup.get(blockHash) match {
+        case Some(block) => block.pure[F]
+        case None =>
+          val blockHashString = Base16.encode(blockHash.toByteArray)
+          Sync[F].raiseError[BlockMetadata](
+            new Exception(s"Missing block hash $blockHashString in block dag.")
+          )
+      }
```

Comment:
this is the same as the code below but the match is more error prone than using the standard operators like `map` and `getOrElse`

```scala
    def unsafeLookup(blockHash: BlockHash): F[BlockMetadata] =
      dataLookup
        .get(blockHash)
        .map(_.pure[F])
        .getOrElse {
          val blockHashString = Base16.encode(blockHash.toByteArray)
          Sync[F].raiseError[BlockMetadata](
            new Exception(s"Missing block hash $blockHashString in block dag.")
          )
      }
```

Reply:
Hmm I find this code harder to read...

I'm trying to imagine myself as someone who just learned Scala: .map on an Option and _.pure[F] are confusing concepts imo.

Reply:
```scala
      dataLookup.get(blockHash) match {
        case Some(block) => block.pure[F]
        case None =>
          val blockHashString = Base16.encode(blockHash.toByteArray)
          Sync[F].raiseError[BlockMetadata](
            new Exception(s"Missing block hash $blockHashString in block dag.")
          )
      }
```

Here is how I read this code:

1. `dataLookup.get(blockHash)` - "Oh we lookup a map"
2. `case Some(block) =>` vs `case None =>` - "Maybe the block exists, if it does, go on with this case. Otherwise go on with the None case"
3. `=> block.pure[F]` - "Ok I have no idea what this .pure[F] thing does but I'm assuming it is some minor modification to block, but I'm assuming the general idea is that when the block exists, I just return it"
4. 
```
          val blockHashString = Base16.encode(blockHash.toByteArray)
          Sync[F].raiseError[BlockMetadata](
            new Exception(s"Missing block hash $blockHashString in block dag.")
          )
``` 
"Ok so there is this weird `Sync[F].raiseError` syntax but from the name I guess it throws some kind of error - and this makes sense, when the blockHash doesn't exist we need to throw an error"

vs

```scala
    def unsafeLookup(blockHash: BlockHash): F[BlockMetadata] =
      dataLookup
        .get(blockHash)
        .map(_.pure[F])
        .getOrElse {
          val blockHashString = Base16.encode(blockHash.toByteArray)
          Sync[F].raiseError[BlockMetadata](
            new Exception(s"Missing block hash $blockHashString in block dag.")
          )
      }
```

1. `dataLookup.get(blockHash)`- "Oh we lookup a map"
2. `.map(_.pure[F])` - "We map over the result? Does it return a list? What is this magical underscore pure thing? Searches for an hour to learn that you can map over Options or any Monad. And then spends another day reading up all the tutorials on Monads."